### PR TITLE
resolve nested entities

### PR DESCRIPTION
**Pull request by ericcater**
You can resolve nested entities via the graphql api.

For example

```typescript
findManySurveys {
    id
    orgs {
      id
    }
  }
```

this gets all id's for orgs attach to a survey (and shows which survey they're attached too).

Optimized for service -> database queries. see [prisma fluent api](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#do-i-have-to-use-the-fluent-api-to-enable-batching-of-queries) for details

e2e pass

### CODE REVIEW

File: apps/ods-api/src/user/user.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Resolver, Query, Parent, ResolveField } from '@nestjs/graphql';
```
Comment (yurisim) [inline apps/ods-api/src/user/user.resolver.ts:1]: Did you mean to include these? There's no module file for these.

File: apps/ods-api/src/survey/survey.resolver.ts [hunk 2]
```diff
@@ -51,7 +60,24 @@ export class SurveyResolver {
   async delete(
     @Args('surveyWhereUniqueInput')
     surveyWhereUniqueInput: SurveyWhereUniqueInput
-  ): Promise<{ deleted: boolean }> {
+  ) {
     return this.surveyService.delete(surveyWhereUniqueInput);
   }
+
+  @ResolveField(() => [OrgGQL], { name: 'orgs' })
```
Comment (yurisim) [inline apps/ods-api/src/survey/survey.resolver.ts:67]: Did you intend to put a { name } field on the othe rproperties in the file? I think they default to just the name of the method, in which case {name: 'orgs'} is redundant

File: apps/ods-api/src/surveyResponse/surveyResponse.resolver.ts [hunk 3]
```diff
@@ -15,23 +24,30 @@ import {
...
+  ): Promise<SurveyResponseGQL | null> {
     return this.surveyResponseService.findUnique(
       surveyResponseWhereUniqueInput
     );
   }
 
   @Mutation(() => SurveyResponseGQL, { name: 'createSurveyResponse' })
   // @UseGuards(AccessTokenAuthGuard)
-  async create(
+  create(
```
Comment (yurisim) [inline apps/ods-api/src/surveyResponse/surveyResponse.resolver.ts:47]: Was async accidently removed? or was this purposeful? Which branch was it did you make resolver methods require async?

File: apps/ods-api/src/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
```
Comment (yurisim) [inline apps/ods-api/src/user/user.service.ts:1]: Same comment as the resolver

File: libs/types/ods/src/entity/org.entity.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
-import { ObjectType, Field, InputType } from '@nestjs/graphql';
+import { ObjectType, Field, HideField } from '@nestjs/graphql';
 import { Org, OrgTier } from '.prisma/ods/client';
 
 @ObjectType()
-@InputType('OrgGQLInput')
```
Comment (yurisim) [inline libs/types/ods/src/entity/org.entity.ts:5]: What's the significant of removing the Input types from these files?

**Review by yurisim — CHANGES_REQUESTED** [review]

**Comment by yurisim** [comment]
Overall this is a LOT more readable than the last batch of PR. This is great! :)

File: apps/ods-api/src/survey/survey.resolver.ts (hunk 2 above)
Comment (ericcater) [reply apps/ods-api/src/survey/survey.resolver.ts:67]: It is redundant but when I was implementing things it was giving more errors if I mistyped something. But I think that was when I was doing it a bit different. So on the go around I meant to remove them all.

**Review by yurisim — APPROVED** [review]

**Pull request merged**